discovery/kubernetes/ingress_test.go

```diff
@@ -200,3 +219,49 @@ func TestIngressDiscoveryNamespaces(t *testing.T) {
 		expectedRes:      expected,
 	}.Run(t)
 }
+
+func TestMatchHostnamePattern(t *testing.T) {
```

Comment:
maybe we want to remove this test before a merge but I just added it for completeness